cocos/renderer/CCTextureCache.cpp

```diff
@@ -102,6 +102,12 @@ void TextureCache::addImageAsync(const std::string &path, const std::function<vo
 
     std::string fullpath = FileUtils::getInstance()->fullPathForFilename(path);
 
+    // check if file exists
+    if (! FileUtils::getInstance()->isFileExist( fullpath )) {
```

Comment:
`fullpath` may contain data if `path` was absolute even if it doesn't exist, so line 106 should be

`if (fullpath.empty() || ! FileUtils::getInstance()->isFileExist( fullpath )) {`

A later fail by returning nullptr already happens.  During `addImageAsyncCallBack()` if `image` fails to load then `texture` stays `nullptr` and is returned regardless at `asyncStruct->callback(texture);`